vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1295,6 +1298,33 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private boolean hasSiblingWithSameType(Node child) throws RepositoryException {
+
+        Node parent = child.getParent();
+
+        try {
+            EffectiveNodeType ent = EffectiveNodeType.ofNode(parent);
+            String typeName = ent.getApplicableChildNodeDefinition(child.getName(), child.getPrimaryNodeType()).get().getName();
+
+            NodeIterator iter = parent.getNodes();
+            while (iter.hasNext()) {
+                Node sibling = iter.nextNode();
+                if (!sibling.isSame(child)) {
+                    Optional<NodeDefinition> childDef = ent.getApplicableChildNodeDefinition(sibling.getName(),
+                            sibling.getPrimaryNodeType());
```

Comment:
consider mixins

Reply:
How are mixins relevant here?

Reply:
See above, if the parent enforces a child of a certain mixin this will only be returned by `getApplicableChildNodeDefinition` if the mixins are passed as well.